packages/zoe/src/zoe.js

```diff
@@ -635,261 +410,270 @@ function makeZoe(additionalEndowments = {}, vatPowers = {}) {
   // an object with a complete method for leaving the contract on demand.
 
   /** @type {ZoeService} */
-  const zoeService = harden(
+  const zoeService = harden({
+    getInviteIssuer: () => inviteIssuer,
+
     /**
-     * @param {any} code
+     * Create an installation by permanently storing the bundle. It will be
+     * evaluated each time it is used to make a new instance of a contract.
+     * We have a moduleFormat to allow for different future formats without
+     * silent failures.
      */
-    {
-      getInviteIssuer: () => inviteIssuer,
-
-      /**
-       * Create an installation by safely evaluating the code and
-       * registering it with Zoe. We have a moduleFormat to allow for
-       * different future formats without silent failures.
-       */
-      install: async bundle => {
-        const installation = await evalContractBundle(
-          bundle,
-          additionalEndowments,
-          vatPowers,
-        );
-        const installationHandle = installationTable.create(
-          harden({ installation, bundle }),
-        );
-        return installationHandle;
-      },
+    install: async bundle => {
+      const installationHandle = harden({});
+      installationTable.create(harden({ bundle }), installationHandle);
+      return installationHandle;
+    },
 
-      /**
-       * Makes a contract instance from an installation and returns a
-       * unique handle for the instance that can be shared, as well as
-       * other information, such as the terms used in the instance.
-       * @param  {object} installationHandle - the unique handle for the
-       * installation
-       * @param {Object.<string,Issuer>} issuerKeywordRecord - a record mapping
-       * keyword keys to issuer values
-       * @param  {object} terms - optional, arguments to the contract. These
-       * arguments depend on the contract.
-       */
-      makeInstance: (
-        installationHandle,
-        issuerKeywordRecord = harden({}),
-        terms = harden({}),
-      ) => {
-        assert(
-          installationTable.has(installationHandle),
-          details`${installationHandle} was not a valid installationHandle`,
-        );
-        const { installation } = installationTable.get(installationHandle);
-        const instanceHandle = harden({});
-        const contractFacet = makeContractFacet(instanceHandle);
+    /**
+     * Makes a contract instance from an installation and returns the
+     * invitation and InstanceRecord.
+     *
+     * @param  {object} installationHandle - the unique handle for the
+     * installation
+     * @param {Object.<string,Issuer>} issuerKeywordRecord - a record mapping
+     * keyword keys to issuer values
+     * @param  {object} terms - optional, arguments to the contract. These
+     * arguments depend on the contract.
+     * @returns {MakeInstanceResult}
```

Comment:
Since we're calling `E()` I think it's actually a promise for a MakeInstanceResult
```suggestion
     * @returns {Promise<MakeInstanceResult>}
```

Reply:
You're right, but it's indistinguishable, since all callers (other than tests) have to use `E(zoe).makeInstance(...)` anyway.  

Do you think it would be helpful to add `Promise<>` here? My instinct is no.